### PR TITLE
Remove references to "new" features introduced before C# 9

### LINKED ISSUE #30422: Remove "Beginning in C#" phrase for older versions
Throughout the C# guide, there are phrases like "Beginning in C# N", or simply "In C# N".

In some cases, the version referenced goes all the way back to C# 3.  

I think we should remove this phrasing for all versions prior to C# 9. (Effectively keep 9, 10, and 11). Then, continue rolling forward to keep only the last 3 versions.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #30422

Several locations in our docs refer to features as "new" and reference the version when introduced. This is great for the newest features, but doesn't enhance the experience for features introduced in versions as old as C# 2.

Each commit in this PR goes through C# versions starting with C# 2.0, removing language that references when a feature was introduced. The latest version where this language was removed is C# 8.0.

The exceptions are articles on compiler errors and warnings, where version differences are important, and articles that are specifically about C# version history.

Note for reviewers:

There are a lot of files in this PR. For that reason, I didn't address any other issues or editing concerns beyond the notes on "new" in a certain version.

Reviewing this commit-by-commit may be quickest. Each successive commit removes notes about a single C# version starting with Verson 2 (Yes, there were notes about "new" features almost 20 years ago.)

The last two commits update snippet config files and add notes about error suppression in our snippets build because the container doesn't have the Office assemblies installed.

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening.

File: samples/snippets/csharp/VS_Snippets_VBCSharp/csprogguideindexedproperties/cs/snippets.5000.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "host": "visualstudio",
+    "expectederrors": [
+        {
+            "file": "docs/samples/snippets/csharp/VS_Snippets_VBCSharp/csprogguideindexedproperties/cs/Program.cs",
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_VBCSharp/csprogguideindexedproperties/cs/snippets.5000.json:5]:
```suggestion
            "file": "samples/snippets/csharp/VS_Snippets_VBCSharp/csprogguideindexedproperties/cs/Program.cs",
```

**Comment by BillWagner** [comment]
closing and reopening to try the new snippets build

File: docs/csharp/language-reference/operators/delegate-operator.md [hunk 2]
```diff
@@ -12,12 +12,11 @@ The `delegate` operator creates an anonymous method that can be converted to a d
...
-> Beginning with C# 3, lambda expressions provide a more concise and expressive way to create an anonymous function. Use the [=> operator](lambda-operator.md) to construct a lambda expression:
->
-> [!code-csharp-interactive[lambda expression](snippets/shared/DelegateOperator.cs#Lambda)]
->
-> For more information about features of lambda expressions, for example, capturing outer variables, see [Lambda expressions](lambda-expressions.md).
+Lambda expressions provide a more concise and expressive way to create an anonymous function. Use the [=> operator](lambda-operator.md) to construct a lambda expression:
+
+[!code-csharp-interactive[lambda expression](snippets/shared/DelegateOperator.cs#Lambda)]
+
+For more information about features of lambda expressions, for example, capturing outer variables, see [Lambda expressions](lambda-expressions.md).
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/delegate-operator.md:19]: I think it's still good to keep the NOTE block to highlight the lambda expressions.

File: docs/csharp/language-reference/statements/selection-statements.md [hunk 3]
```diff
@@ -79,25 +79,16 @@ A case pattern may be not expressive enough to specify the condition for the exe
...
-
 ## C# language specification
 
 For more information, see the following sections of the [C# language specification](~/_csharpstandard/standard/README.md):
 
 - [The `if` statement](~/_csharpstandard/standard/statements.md#1282-the-if-statement)
 - [The `switch` statement](~/_csharpstandard/standard/statements.md#1283-the-switch-statement)
 
-For more information about features introduced in C# 7.0 and later, see the following feature proposal notes:
+For more information about pattern matching switch statement, see the following feature proposal notes:
```
Comment (pkulikov) [inline docs/csharp/language-reference/statements/selection-statements.md:89]:
```suggestion
For more information about pattern matching `switch` statement, see the following feature proposal notes:
```

File: docs/csharp/language-reference/operators/delegate-operator.md (hunk 2 above)
Comment (BillWagner) [inline docs/csharp/language-reference/operators/delegate-operator.md:19]:
```suggestion
> [!NOTE]
> Lambda expressions provide a more concise and expressive way to create an anonymous function. Use the [=> operator](lambda-operator.md) to construct a lambda expression:
>
> [!code-csharp-interactive[lambda expression](snippets/shared/DelegateOperator.cs#Lambda)]
>
> For more information about features of lambda expressions, for example, capturing outer variables, see [Lambda expressions](lambda-expressions.md).
```

**Review by IEvangelist — APPROVED** [review]
Admittedly, I skimmed a lot of this... but it looks good to me. :shipit: when you're ready.

**Pull request merged**